velox/vector/tests/VectorTest.cpp

```diff
@@ -3101,5 +3101,179 @@ TEST_F(VectorTest, appendNulls) {
   // Append negative.
   EXPECT_ANY_THROW(rowVector->appendNulls(-1));
 }
+
+TEST_F(VectorTest, primitiveTypeNullEqual) {
+  auto base = makeNullableFlatVector<int32_t>({1, 2, std::nullopt});
+  auto other = makeNullableFlatVector<int32_t>({1, std::nullopt, 1});
+
+  auto equalHasValue = [&](vector_size_t i,
+                           vector_size_t j,
+                           CompareFlags::NullHandlingMode nullHandlingMode,
+                           bool expected) {
+    ASSERT_EQ(
```

Comment:
It is better to keep ASSERT_XXX calls outside of the lambda so that failures point to the right line in the test, not to the same line. See example I shared earlier.

Reply:
>keep ASSERT_XXX calls outside of the lambda so that failures point to the right line in the test, not to the same line.

Got it, I remember that I used to encounter this situation before. I've updated the test cases according to this principle. Thanks.